samples/mysql-schema/src/main/java/io/javaoperatorsdk/operator/sample/MySQLSchemaOperator.java

```diff
@@ -24,7 +24,7 @@ public static void main(String[] args) throws IOException {
     Config config = new ConfigBuilder().withNamespace(null).build();
     KubernetesClient client = new DefaultKubernetesClient(config);
     Operator operator = new Operator(client, DefaultConfigurationService.instance());
-    operator.registerControllerForAllNamespaces(new SchemaController(client));
+    operator.register(new SchemaController(client));
```

Comment:
How does user can configure the namespaces in runtime then?
Note that whatever is going to be implemented should work for pure-java operators as well.

Reply:
This is done by using `namespaces = ControllerConfiguration.WATCH_ALL_NAMESPACES_MARKER` on the `Controller` annotation.

Reply:
That’s not changale in runtime,
In pure java operators it still should be possible for user to get the namespaces some how and respct them in registration.

Reply:
This should be overridden via externalized configuration.

Reply:
Not sure what external configuration for pure-java applications mean!
Isn’t that too opinionated? Also do we have this possibility already?

I’m asking because I’m using the removed methods to configure my controllers already and there may be other users who use them, so we need to offer a reasonable alternative before removing them

Reply:
Isn't that contradictory though? If you use this method, then, by definition, you cannot change it at runtime. In this case, using the annotation field should work just fine, no? See for example: https://github.com/java-operator-sdk/java-operator-sdk/pull/303/files#diff-e0b2d5ca6dfbfa8a591df7661a794ce4bedb24ca5d360ee723c1d76dace47657R27

Reply:
This is not the only method removed from the API, I'm talking about the entire change we have in operator methods, 
with a combination of the, it's already possible to enable controller in cherry-picked namespaces or in **all** namespaces together.

Reply:
Could you please detail a scenario where removing the methods and using the annotation instead wouldn't work? 

The other aspect of this is that putting the configuration on the controller implementation makes it easier to reason because the controller explicitly states how it's supposed to work (watching one namespace, all/some namespaces, cluster-scoped) as opposed to possibly registering a controller in a context where it's not supposed to be used (though I guess, the later is also a risk with externalized configuration).

Reply:
User 1 wants to run operatopr **OP1** just for namespace **ns1** 
User 2 wants to run operatopr **OP1** for namespaces **ns1** and **ns2** 
User 3 wants to run operatopr **OP1** for **all**  the namespaces 

The operator is in pure-java and is deciding which namespaces are in the interest of the operator via command-line arguments. 

So it either calls `registerControllerForAllNamespaces` method if **ALL** is sent as argument method or call `registerController` with the list of given namespaces. 

Now the question is how the same thing can be achieved after removing these methods?

Reply:
That makes sense. How about a `register(controller, overrideConfiguration)` method? It's probably more complex to implement than keeping the separate methods but the issue is that we should always go through the configuration to register controllers because that also has impacts on the event sources and it's not done consistently everywhere.

Reply:
Yep it makes sense to have a configuration class with convenient to use method to let the user change the stuff that are useful to be configured in runtime like: retry, namespaces, cluster scoped,...

Reply:
Ok, I'll get on that then! 😄

Reply:
This is now done.